src/CoreWCF.Kafka/src/CoreWCF/Channels/KafkaMessageProperty.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Confluent.Kafka;
+
+namespace CoreWCF.Channels;
+
+public sealed class KafkaMessageProperty : IMessageProperty
+{
+    private readonly IList<KafkaMessageHeader> _headers = new List<KafkaMessageHeader>();
+
+    public static readonly string Name = "kafkaMessage";
+
+    internal KafkaMessageProperty(ConsumeResult<byte[], byte[]> consumeResult)
+    {
+        foreach (IHeader messageHeader in consumeResult.Message.Headers)
+        {
+            _headers.Add(new KafkaMessageHeader(messageHeader.Key, messageHeader.GetValueBytes()));
+        }
+
+        PartitionKey = consumeResult.Message.Key;
+        Topic = consumeResult.Topic;
+    }
+
+    public IMessageProperty CreateCopy() => this;
+
```

Comment:
If you are going to just return `this`, then drop implementing the interface. This is used when adding a property to clone it if needed so if something in the original is disposed, the copy is still intact. When a message property doesn't implement `IMessageProperty`, it just stores the property as is, like you're doing.

Reply:
done